### PR TITLE
is_zero is a compile-time check if it's the constant zero

### PR DESCRIPTION
**Pull request by abadams**
I think for this code we want a runtime check if it's zero or not.

@rootjalex

### CODE REVIEW

**Review by vksnk — APPROVED** [review]

File: src/Bounds.cpp [hunk 1]
```diff
@@ -653,7 +653,7 @@ class Bounds : public IRVisitor {
                 interval.max = b.max - make_one(t);
             } else if (b.max.type().is_uint()) {
                 // if b.max = 0 then result is [0, 0], else [0, b.max - 1]
-                interval.max = select(is_zero(b.max), make_zero(t), b.max - make_one(t));
+                interval.max = select(b.max == make_zero(t), make_zero(t), b.max - make_one(t));
```
Comment (steven-johnson) [inline src/Bounds.cpp:656]: I had to think a bit before realizing the nature of the change. Maybe we should rename `is_zero` and friends to make this more clear?

Comment (abadams) [reply src/Bounds.cpp:656]: Yeah... maybe is_const_zero?

Comment (abadams) [reply src/Bounds.cpp:656]: (and is_const_one)

Comment (steven-johnson) [reply src/Bounds.cpp:656]: SGTM

**Comment by rootjalex** [comment]
Yes, that is what we want. My bad, I should have checked is_zero more closely.

**Pull request merged**